### PR TITLE
[server] Removes unused RepositoryService.canInstall

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description
PR to remove unused code.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at ac7fe43</samp>

This pull request simplifies and cleans up the code of various classes and files related to prebuilds and repository services. It removes unused dependencies, methods, imports, and tests, and refactors some constructors and helper methods. The overall purpose is to improve code quality, consistency, and performance.

</details>

## Related Issue(s)

Fixes #

## How to test
* See you can create a workspace in preview environment w/o issues.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-remove-f7bcae44fe</li>
	<li><b>🔗 URL</b> - <a href="https://at-remove-f7bcae44fe.preview.gitpod-dev.com/workspaces" target="_blank">at-remove-f7bcae44fe.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - at-remove-unused-caninstall-gha.20553</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-at-remove-f7bcae44fe%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -1122,30 +1120,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 context = WithDefaultConfig.mark(context);
             }
 
-            // if this is an explicit prebuild, check if the user wants to install an app.
```
Comment (AlexTugarev) [inline components/server/src/workspace/gitpod-server-impl.ts:1125]: This is no longer supported for quite a while, i.e. `context` may never get parsed to `StartPrebuildContext` and reach `createWorkspace` here.

`StartPrebuildContext` is used to encapsulate a rich context in prebuild manager, which then uses this as input for workspace starter.

**Review by geropl — APPROVED** [review]
Code changes LGTM, :heavy_check_mark: to unblock.

I'm missing the context atm to verify [this](https://github.com/gitpod-io/gitpod/pull/19112/files#r1401636091), but it has been on the short list for removal for quiiiiite some time now :orange_heart:

File: components/server/src/workspace/gitpod-server-impl.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/server/src/workspace/gitpod-server-impl.ts:1125]: > I'm missing the context atm to verify this

That can be taken quite literally ;-) 

`https://gitpod.io/new/#prebuild/https://github.com/gitpod-io/gitpod`

☝🏻 this results in that 👇🏻 

```
We're sorry, there seems to have been an error.
Error: Couldn't parse context 'prebuild/https://github.com/gitpod-io/gitpod'
```

But in `createWorkspace` the cold path was never removed. 🤷🏻‍♂️

Comment (AlexTugarev) [reply components/server/src/workspace/gitpod-server-impl.ts:1125]: Quick test on the preview env: 
https://at-remove-f7bcae44fe.preview.gitpod-dev.com/new/#https://github.com/AlexTugarev/bel ✔️ (workspace is starting fine)

**Pull request merged**